plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -538,12 +538,12 @@ public function get_field_location( $field_key ) {
 	 *
 	 * @param string           $key          The key of the field.
 	 * @param mixed            $field_value  The value of the field.
-	 * @param array            $field_schema The schema of the field.
 	 * @param \WP_REST_Request $request      The current API Request.
```

Comment:
What is the use case for `$request`? I see it is only passed to the filter hook but its not consumed. I'm wondering if its needed because if there was not $request, this method could be used in other contexts e.g. validating fields on the my account page.

Reply:
The purpose of sending `$request` is so that the values of other fields can be checked too, for example requiring the shopper to enter a value twice to improve accuracy, I don't think there's another way we can get the values of the other additional fields in this hook without it.
<img width="539" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/164bf823-2995-435c-bfc9-d2c9d7192ff8">

Reply:
Would this then be incompatible with fields rendered on the edit order screen/my account pages? The data would be POSTed but not part of a request, currently.

Reply:
I think so then, so that kind of ruins the validation plans for those areas. We do need access to the other fields. I wonder if we could just create the array of other fields ourselves and pass that through rather than the WP_Rest_Request 🤔  let me try with that

Reply:
We agreed to leave request in place and change how its consumed from other locations.